tests/github-retag-and-push.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+
+function finish() {
+    docker stop registry
+}
+
+trap finish EXIT
+
+docker run --detach --publish 5000:5000 --rm --name registry registry:2
+
+docker pull busybox
+
+# valid input
+docker tag busybox libero/my-dummy-project:12345678
+GITHUB_REF=refs/heads/master \
+GITHUB_SHA=12345678 \
+DOCKER_REGISTRY=localhost:5000/ \
+DOCKER_USERNAME=foo \
+DOCKER_PASSWORD=bar \
+github/retag-and-push.sh my-dummy-project 12345678
```

Comment:
wondering what we could assert? Trying to pull the tags that have been generated and pushed?

Reply:
I've implemented some assertions for `retag-and-push.sh`. Migrated the tests for `push-image.sh` but didn't extend them. Could be done in different PR?